### ISSUE TITLE
Updated code to Nextjs 12, firebase v9, and react 18

### ISSUE THREAD

**Issue by dan-sproutward** [mariesta/nextjs-auth-with-firebase#1]
Would you be interested in the update? I can create a PR.

**Comment by mariesta** [comment]
Hey Dan! I agree this repository could use some dusting. I haven't because I would also like to update the article using it. However, to do so, I need to work get permission from Log Rocket first. Considering this article will soon be a year old, I think  I can get my editor to agree on an edit. 🤞 

If it is not too much of a bother, feel free to send a PR update. I won't merge it right away, but it will be helpful when I update the article.

Have a great day!

**Comment by dan-sproutward** [comment]
Will do! And thanks again for all the work in the articles. Articles like yours are very helpful!

**Comment by dan-sproutward** [comment]
Reach out if you have any questions.  

Updating firebase v9 documentation suggests importing auth functions directly so signInWithEmailAndPassword, createUserWithEmailAndPassword, and signOut functions in previous code had to be renamed because they conflicted with the imported function name. 

firebase.js only returns app currently; which may be a result of a limitation in my understanding as I was working through the update and learning firebase, react, and nextjs simultaneously. As a result, auth is retrieved in useFirebaseAuth where all the other functions off auth were used but I presume auth could be returned from the context vs app.

SignOut had an issue where it was looking for loading false and authUser false which is true when an unauthenticated user tries to navigate to logged_in directly but signOut logic was setting loading to true so the redirect never triggered.

I added Google login because I was interested in testing it but as a result I consolidated close out of auth function returns in useFireBaseAuth and I pulled in signIn use of router here; I wasn't totally consistent as I left in the useEffect use of router in logged_in.jsx but that is required to handle direct navigation to that page by an unauthenticated user. I suspect this is not necessary but this centralized handling all auth related logic in one module. I did not understand how to use router with Google login in logged_in.jsx directly in the return "markup" but now that I think of it I could of created a JS function called from the markup and called router from there; I am not sure what the return is called from react that has markup and JS mixed but I like it :).

Hope this info helps consume the PR. Again, thanks for publishing a useful project. I love the internets.
